packages/fxa-payments-server/server/lib/amplitude.js

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const {
+  GROUPS,
+  initialize,
+  mapBrowser,
+  mapFormFactor,
+  mapLocation,
+  mapOs,
+  mapUserAgentProperties,
+  toSnakeCase,
+} = require('../../../fxa-shared/metrics/amplitude.js');
+const config = require('../config');
+const amplitude = config.get('amplitude');
+const joi = require('joi');
+const log = require('./logging/log')();
+const ua = require('../../../fxa-shared/metrics/user-agent');
+const { version: VERSION } = require('../../package.json');
+
+const FUZZY_EVENTS = new Map([
+  [
+    // Emit events from the front-end as `amplitude.${GROUP}.${EVENT}`
+    /^amplitude\.([\w-]+)\.([\w-]+)$/,
+    {
+      group: group => GROUPS[group],
+      event: (group, event) => toSnakeCase(event),
+    },
+  ],
+]);
+
+const transform = initialize({}, {}, FUZZY_EVENTS);
```

Comment:
Rather than call `initialize` here, you might do some dependency injection to make testing easier.

That is, call this `initialize` in post-metrics.js and pass the `transform` result into the function exported below when it's called in the route. Then, you can create a mock `transform` in testing without needing to try mocking the whole module.

Reply:
Great idea! I'll give that a try